src/main/java/co/rsk/federate/config/PowHSMBookkeepingConfig.java

```diff
@@ -49,10 +52,36 @@ public PowHSMBookkeepingConfig(SignerConfig signerConfig, String networkParamete
         this.networkParameter = networkParameter;
     }
 
-    public BigInteger getDifficultyTarget() {
-        return signerConfig.hasPath(DIFFICULTY_TARGET_PATH)
-            ? new BigInteger(signerConfig.getString(DIFFICULTY_TARGET_PATH))
-            : DIFFICULTY_TARGET_DEFAULT;
+    /**
+     * Retrieves the difficulty target needed for the internal PowHSM blockchain.
+     * <p>
+     * The difficulty target can be obtained either from the configuration file or from the PowHSM.
+     * If the difficulty target is specified in the configuration file (indicated by the path {@code DIFFICULTY_TARGET_PATH}),
+     * and if the HSM client version is less than 3, the difficulty target will be read from the configuration file.
+     * Otherwise, the difficulty target will be retrieved from the PowHSM.
+     * </p>
+     *
+     * @param hsmClient the PowHSM client used to retrieve blockchain parameters.
+     * @return the difficulty target as a {@link BigInteger}.
+     * @throws HSMClientException if there is an error communicating with the PowHSM.
+     * @throws IllegalStateException if the difficulty target cannot be read from the configuration file or
+     *                               if it cannot be retrieved from the PowHSM.
+     */
+    public BigInteger getDifficultyTarget(HSMBookkeepingClient hsmClient)
+        throws HSMClientException {
+      if (hsmClient.getVersion() < 3) {
+          return Optional.ofNullable(signerConfig.getString(DIFFICULTY_TARGET_PATH))
+              .map(BigInteger::new)
+              .orElseThrow(() -> 
+                  new IllegalStateException("Unable to read difficulty target from configuration file"));
+      }
+
+      logger.trace("[getDifficultyTarget] Retrieve minimum difficulty target from the PowHSM");
+
+      return Optional.ofNullable(hsmClient.getBlockchainParameters())
+          .map(PowHSMBlockchainParameters::getMinimumDifficulty)
+          .orElseThrow(() ->
+              new IllegalStateException("Unable to retrieve difficulty target using PowHSM"));
```

Comment:
```suggestion
     try {
          logger.trace("[getDifficultyTarget] Retrieve minimum difficulty target from the PowHSM");
          
      return Optional.ofNullable(hsmClient.getBlockchainParameters())
          .map(PowHSMBlockchainParameters::getMinimumDifficulty)
     } catch(HSMUnsupportedTypeException) (
         logger(try get the value from config file)
          return Optional.ofNullable(signerConfig.getString(DIFFICULTY_TARGET_PATH))
              .map(BigInteger::new)
              .orElseThrow(() -> 
                  new IllegalStateException("Unable to read difficulty target from configuration file"));
      } 
```
What do you think?

Reply:
I agree, I have updated the method. I have also added in the catch statement the `IllegaStateException` as a fallback in case the call to the PowHSM fails with an invalid value of some sort.